react-restaurant-review-app/src/components/Login/Login.tsx

```diff
@@ -3,44 +3,43 @@ import styles from "./Login.module.css";
 import restaurnatImage from "../../assets/images/019a0242b61b695b45ca70321ca186a0.jfif";
 import { useNavigate } from "react-router-dom";
 import { User } from "../../models/user";
+import { post } from "../../utils/service";
 
 export default function Login() {
-  const [formData, setFormData] = useState({ userName: "", password: "" });
+  const [formData, setFormData] = useState<User>({
+    id: "",
+    password: "",
+    userName: "",
+    role: 2,
+  });
   const [userData, setUserData] = useState<User>({
     id: "",
     password: "",
     userName: "",
-    role: 0,
+    role: 2,
   });
   const navigate = useNavigate();
 
-  function handleFormChange(event: any) {
+  function handleFormChange(event: React.ChangeEvent<HTMLInputElement>) {
     const { name, value } = event.target;
 
     setFormData((prevValue) => {
       return { ...prevValue, [name]: value };
     });
   }
 
-  function handleSubmit(event: React.FormEvent) {
+  async function handleSubmit(event: React.FormEvent) {
     event.preventDefault();
+    let response = await post<User, { user: User; token: string }>(
+      "login",
+      formData
+    );
 
-    fetch("http://localhost:8080/login", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(formData),
-    })
-      .then((res) => res.json())
-      .then((res) => {
-        if (res.isSuccess) {
-          setUserData(res.payLoad.user);
-          navigate("/restaurants");
-          localStorage.setItem("token", res.payLoad.token);
-        }
-      })
-      .catch((error) => console.error("Error:", error));
+    if (response) {
+      setUserData(response.user);
+      navigate("/restaurants");
+      localStorage.setItem("token", response.token);
```

Comment:
Lets set the token before navigating to make sure it is set before we move to the next page